### PR TITLE
added sbpiper and sbpipe recipes

### PR DESCRIPTION
**Pull request by pdp10**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by pdp10** [comment]
I do not understand what the problem is. 
In my local machine `conda-build meta.yaml` for meta.yaml of r-sbpiper runs correctly. 

Any help would be appreciated!

**Comment by daler** [comment]
@pdp10 this recipe uses the conda-build 3 syntax, which we are not yet supporting yet. We're close! But not yet. Try replacing `host:` with `build:`.

And to verify locally, please use https://bioconda.github.io/contribute-a-recipe.html#bootstrap-method as it mimics the CircleCI setup.

**Comment by pdp10** [comment]
Thank you for your help, @daler .
I still have these major errors: 
```
....19:19:01 BIOCONDA ERROR COMMAND FAILED: docker run --net host --rm -v /tmp/tmps82b34z4/build_script.bash:/opt/build_script.bash -v /home/circleci/project/miniconda/conda-bld:/opt/host-conda-bld -v /home/circleci/project/recipes/r-sbpiper:/opt/recipe -e LANG=en_US.UTF-8 -e CONDA_ROOT=/home/circleci/project/miniconda -e CONDA_ALLOW_SOFTLINKS=false -e CONDA_PY=27 -e CONDA_HTSLIB=1.7 -e CONDA_BOOST=1.64 -e CONDA_R=3.4.1 -e CONDA_PERL=5.22.0 -e CONDA_NPY=112 -e CONDA_NCURSES=5.9 -e CONDA_GSL=1.16 -e CONDA_GMP=5.1 -e CONDA_HDF5=1.8.17 -e MACOSX_DEPLOYMENT_TARGET=10.9 -e CONDA_ZLIB=1.2.11 -e CONDA_BZIP2=1.0 -e CONDA_XZ=5.2 -e CONDA_BAMTOOLS=2.4.1 -e HOST_USER_ID=1001 tmp-bioconda-builder /bin/bash /opt/build_script.bash
19:19:01 BIOCONDA ERROR STDOUT+STDERR:
bash: cannot set terminal process group (-1): Inappropriate ioctl for device
bash: no job control in this shell
``` 

My meta.yaml files were generated via `conda skeleton` from cran (sbpiper) and pypi (sbpipe).

**Comment by daler** [comment]
@pdp10 it appears that `conda skeleton cran` has changed recently, and on our CircleCI infrastructure it looks like [this clause]( https://github.com/bioconda/bioconda-recipes/pull/8810/files#diff-b8fb0688fd30d692a38632a9cdc04adfR3) in `build.sh` is evaluating to True.

I would try mimicking an existing build.sh from a bioconda package, e.g. https://github.com/bioconda/bioconda-recipes/blob/master/recipes/r-bcbiobase/build.sh.

The text you're seeing -- `bash: cannot set terminal process group (-1): Inappropriate ioctl for device
bash: no job control in this shell` -- is normal and is an artifact of the docker container we're running. The error that caused the build failure is 

```
> library('sbpiper')
Error in library("sbpiper") : ‘sbpiper’ is not a valid installed package
Execution halted
```

**Comment by pdp10** [comment]
@daler Thank you very much for your help!

File: recipes/r-sbpiper/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+$R CMD INSTALL --build .
+
+
+## conda-build 3
```
Comment (daler) [inline recipes/r-sbpiper/build.sh]: rm these comments please

File: recipes/r-sbpiper/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{% set version = '1.8.0' %}
+
+{% set posix = 'm2-' if win else '' %}
```
Comment (daler) [inline recipes/r-sbpiper/meta.yaml]: please rm any windows-related lines, as we don't build for windows

File: recipes/r-sbpiper/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+source:
+  fn: sbpiper_1.8.0.tar.gz
+  url:
+    - https://cran.r-project.org/src/contrib/sbpiper_1.8.0.tar.gz
+    - https://cran.r-project.org/src/contrib/Archive/sbpiper/sbpiper_1.8.0.tar.gz
+  sha256: c619dc59a6e32e28951b3f0e65a682b27d8890fc6bdedd316f0417a7190d9115
+
+build:
+  merge_build_host: True  # [win]
```
Comment (daler) [inline recipes/r-sbpiper/meta.yaml]: rm this too

File: recipes/r-sbpiper/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    - r-data.table
+    - r-factoextra
+    - r-ggplot2 >=2.2.0
+    - r-reshape2
+    - r-scales
+    - r-stringr
+
+test:
+  commands:
+    - $R -e "library('sbpiper')"           # [not win]
```
Comment (daler) [inline recipes/r-sbpiper/meta.yaml]: should just be:
```yaml
- $R -e "library('sbpiper')" 
```

**Comment by pdp10** [comment]
Thanks again, @daler ! The comments have been removed.

**Comment by daler** [comment]
Looks good, thanks!

**Pull request merged**